fluent/utils.py

```diff
@@ -0,0 +1,43 @@
+class cachedproperty(object):
```

Comment:
Is there a value of that over typical lazy override where you define a property as a getter that after initial execution overrides the property with its result?

Reply:
cachedproperty is quite neat in that it has zero overhead after it has run once (not even a function call, due to the way that Python look things up in object/class dicts), plus it is neat to implement. (Maybe that is what you meant by "overriding the property with its result", not sure, cachedproperty simply wraps this behaviour up neatly).

However, I don't think we really need it at all - the only use for this laziness was not really necessary.